src/query/sql/src/planner/binder/location.rs

```diff
@@ -498,12 +518,9 @@ pub async fn parse_uri_location(
                     ),
                 ));
             }
-            l.connection.check().map_err(|_| {
-                Error::new(
-                    ErrorKind::InvalidInput,
-                    anyhow!("connection_name can not be used with other connection options"),
-                )
-            })?;
+            l.connection
```

Comment:
@andylokandy 

the original error is more clear

https://github.com/datafuselabs/databend/actions/runs/9195244220/job/25291136363?pr=15618

```
 >>>> create stage my_stage url= 's3://testbucket/admin/tempdata/' connection = (connection_name='my_conn', endpoint_url='xx');
-Error: APIError: ResponseError with 4000: invalid input parameter (connection_name can not be used with other connection options)
+Error: APIError: ResponseError with 4000: invalid input parameter (connection params invalid: expected [connection_name], got [endpoint_url])
```